doc/services/storage/zms/zms.rst

```diff
@@ -0,0 +1,418 @@
+To make the garbage collector not affect the life expectancy of the device it is recommended
+to dimension correctly the partition size. Its size should be the double of the maximum size of
+data (including extra headers) that could be written in the storage.
+
+See :ref:`free-space`.
+
+How to compute device lifetime
+==============================
+
+Storage devices whether they are classical Flash or new technologies like RRAM/MRAM has a limited
+life expectancy which is determined by the number of times memory cells can be erased/written.
+Flash devices are erased one page at a time as part of their functional behavior (otherwise
+memory cells cannot be overwritten) and for non erasable storage devices memory cells can be
+overwritten directly.
+
+A typical scenario is shown here to calculate the life expectancy of a device.
+Let's suppose that we store an 8 bytes variable using the same ID but its content changes every
+minute. The partition has 4 sectors with 1024 bytes each.
+Each write of the variable requires 16 bytes of storage.
+As we have 944 bytes available for ATEs for each sector, and because ZMS is a fast-forward
+storage system, we are going to rewrite the first location of the first sector after
+(944 * 4) / 16 = 236 minutes.
+
+In addition to the normal writes, garbage collector will move the still valid data from old
+sectors to new ones.
+As we are using the same ID and a big partition size, no data will be moved by the garbage
+collector in this case.
+For storage devices that could be written 20000 times, the storage will last about
+4.720.000 minutes (~9 years).
+
+To make a more general formula we must first compute the effective used size in ZMS by our
+typical set of data.
+For id/data pair with data <= 8 bytes, effective_size is 16 bytes
+For id/data pair with data > 8 bytes, effective_size is 16 bytes + sizeof(data)
+Let's suppose that total_effective_size is the total size of the set of data that is written in
+the storage and that the partition is well dimensioned (double of the effective size) to avoid
+having the garbage collector moving blocks all the time.
+
+The expected life of the device in minutes is computed as ::
+
+(SECTOR_EFFECTIVE_SIZE * SECTOR_NUMBER * MAX_NUM_WRITES) / (TOTAL_EFFECTIVE_SIZE * WR_MIN)
+
+Where:
+
+``SECTOR_EFFECTIVE_SIZE``: is the size sector - header_size(80 bytes)
+
+``SECTOR_NUMBER``: is the number of sectors
+
+``MAX_NUM_WRITES``: is the life expectancy of the storage device in number of writes
+
+``TOTAL_EFFECTIVE_SIZE``: Total effective size of the set of written data
+
+``WR_MIN``: Number of writes of the set of data per minute
+
+ZMS Features
+************
+ZMS has introduced many features compared to existing storage system like NVS and will evolve
+from its initial version to include more features that satisfies new technologies requirements
+such as low latency and bigger storage space.
+
+Existing features
+=================
+Version1
```

Comment:
Is the version present in the on-disk format?

Reply:
Yes there is a field in the header's metadata for that